File: guess_the_number.go
package main

import (
    "fmt"
    "math/rand"
    "time"
)

func main() {
    rand.Seed(time.Now().UnixNano())
    number := rand.Intn(100) + 1

    fmt.Println("Welcome to Guess the Number! I'm thinking of a number between 1 and 100. Can you guess what it is?")

    for {
        var guess int
        fmt.Print("Enter your guess: ")
        fmt.Scan(&guess)

        if guess == number {
            fmt.Println("Congratulations, you guessed the number!")
            break
        } else if guess < number {
            fmt.Println("Too low! Guess again.")
        } else {
            fmt.Println("Too high! Guess again.")
        }
    }
}

File: guess_the_number.py
import random

# generate a random number between 1 and 100
number = random.randint(1, 100)

# welcome message
print("Welcome to Guess the Number! I'm thinking of a number between 1 and 100. Can you guess what it is?")

# loop until the user guesses correctly
while True:
    # ask the user to guess the number
    guess = int(input("Enter your guess: "))
    
    # check if the guess is correct
    if guess == number:
        print("Congratulations, you guessed the number!")
        break
    elif guess < number:
        print("Too low! Guess again.")
    else:
        print("Too high! Guess again.")
